### PR TITLE
add support for record series

### PR DESCRIPTION
**Pull request by Ke-Jin-Laserfiche**

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -330,9 +330,14 @@ any,
...
       await this.repositoryBrowser?.current?.initAsync(this.lfRepoTreeService!, focusedNode);
+      if (this.repositoryBrowser?.current) {
+        this.repositoryBrowser.current.isSelectable = this.isNodeSelectable;
+        this.repositoryBrowser.current.addEventListener('entrySelected', this.onEntrySelected );
+      }
     }
 
-  isNodeSelectable = (node: LfRepoTreeNode) => {
+  isNodeSelectable = async (treenode: LfTreeNode) => {
+    const node = treenode as LfRepoTreeNode;
```
Comment (alexgomezlf) [inline src/App.tsx:340]: RecordSeries should be selectable too?

File: src/App.tsx [hunk 2]
```diff
@@ -365,25 +370,44 @@ any,
...
-          disabled: false,
-          tag: {
-            handler: () => { this.setState({showNewFolderDialog: true}); }
+      const selecteFolder = this.repositoryBrowser.current?.currentFolder as LfRepoTreeNode;
+      if (selecteFolder.entryType === EntryType.RecordSeries) {
+        this.toolBarElement.current.dropdown_options = [
+          {
+            name: this.REFRESH,
+            disabled: false,
+            tag: {
```
Comment (alexgomezlf) [inline src/App.tsx:379]: we should still show option, it should just be disabled

File: src/App.tsx [hunk 3]
```diff
@@ -523,9 +547,15 @@ any,
     await this.loginComponent.current?.refreshTokenAsync(true);
   }
 
-  setShouldShowSelect(): void {
+  async setShouldShowSelect() {
+    if (this.repositoryBrowser?.current?.currentFolder) {
+      const isSelectable = await this.isNodeSelectable(this.repositoryBrowser?.current?.currentFolder);
+      this.setState({
+        shouldShowSelect: !this.entrySelected && isSelectable
```
Comment (alexgomezlf) [inline src/App.tsx:554]: do you need to return after first setState? (or add an else)

File: src/App.tsx (hunk 1 above)
Comment (Ke-Jin-Laserfiche) [reply src/App.tsx:340]: I thought recordseries only has recordfolders or recordseries. can we save a document to a recordseries?

Comment (alexgomezlf) [reply src/App.tsx:340]: try it?

**Review by alexgomezlf — APPROVED** [review]

**Pull request merged**